core/src/main/scala/org/http4s/blaze/channel/nio2/ClientChannelFactory.scala

```diff
@@ -21,7 +23,16 @@ import scala.util.control.NonFatal
 final class ClientChannelFactory(
     bufferSize: Int = DefaultBufferSize,
     group: Option[AsynchronousChannelGroup] = None,
-    channelOptions: ChannelOptions = ChannelOptions.DefaultOptions) {
+    channelOptions: ChannelOptions = ChannelOptions.DefaultOptions,
+    scheduler: TickWheelExecutor = Execution.scheduler,
+    connectingTimeout: Option[FiniteDuration] = None) {
```

Comment:
Why `Option[FiniteDuration]` instead of `Duration`, so no timeout is `Duration.Inf`?  (I wish we could eliminate negative durations in either encoding, but nobody tends to send them in practice.)